src/MySTMarkdownCell.tsx

```diff
@@ -34,13 +51,34 @@ export class MySTMarkdownCell
       const node = document.createElement('div');
       this.myst = { node };
     }
+
+    this._doneRendering = new PromiseDelegate<void>();
     const notebook = this.parent as StaticNotebook;
     this.myst.pre = undefined;
-    parseContent(notebook);
+    const parseComplete = parseContent(notebook);
     const widget = new Widget({ node: this.myst.node });
     widget.addClass('myst');
     this.addClass('jp-MySTMarkdownCell');
     this.inputArea.renderInput(widget);
+    if (parseComplete) {
+      parseComplete.then(() => this._doneRendering.resolve());
+    } else {
+      // Something went wrong, but we will still signal the next step?
+      this._doneRendering.resolve();
```

Comment:
Should we `reject` this in the failure case?

Reply:
Done!